e2e/test/provisioning/ReprovisioningE2ETests.cs

```diff
@@ -542,27 +550,47 @@ private async Task ConfirmDeviceWorksAfterReprovisioning(DeviceRegistrationResul
                 using var testMessage = new Client.Message(Encoding.UTF8.GetBytes("TestMessage"));
                 await iotClient.SendEventAsync(testMessage).ConfigureAwait(false);
 
-                //twin can be configured to revert back to default twin when provisioned, or to keep twin
+                // Twin can be configured to revert back to default twin when provisioned, or to keep twin
                 // from previous hub's records.
-                if (twinOperationsAllowed)
+                if (transportProtocolSupportsTwinOperations)
                 {
                     Twin twin = await iotClient.GetTwinAsync().ConfigureAwait(false);
 
-                    if (reprovisionPolicy.MigrateDeviceData)
-                    {
-                        Assert.AreNotEqual(1, twin.Properties.Desired.Count);
-                        Assert.AreNotEqual(0, twin.Properties.Desired.Version);
-                        Assert.AreEqual(ProvisioningRegistrationSubstatusType.DeviceDataMigrated, result.Substatus);
-                    }
-                    else if (reprovisionPolicy.UpdateHubAssignment)
+                    // Reprovision
+                    if (reprovisionPolicy.UpdateHubAssignment)
```

Comment:
The pervious assertions seemed confusing to me, since the logic relied on `reprovisionPolicy.MigrateDeviceData` and `reprovisionPolicy.UpdateHubAssignment` being mutually exclusive, while they are not.
I reworked the assertions based on the explanation here: https://docs.microsoft.com/en-us/azure/iot-dps/concepts-device-reprovision

@timtay-microsoft I added this in now; could you give this another look? I'd like to get another pair of eyes on this before I merge it in.